scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1232,16 +1238,48 @@
+                ; We want to autoscale any unmatched job
+                ;     OR
+                ; we want to scale our max-jobs-for-autoscaling by the fraction of the jobs we weren't able to just match.
+                ; E.g. If we didn't match 20% of the queue, then we want to autoscale to 20% of max-jobs-for-autoscaling.
+                ; We include a scale factor however, so that if we don't match 20% and scale factor is 2.5, we'll generate
+                ; pods for 50% of max-jobs-for-autoscaling.
+                ; We do this to vary our aggression for autoscaling based on how well we're matching jobs on our existing resources.
+                ; If we're matching most of the jobs in the queue then we don't need to autoscale much. If we are not matching anything
+                ; then we want to autoscale maximally aggressively.
+                max-jobs-for-autoscaling-scaled (-> fraction-unmatched-jobs
+                                                    (* autoscaling-scale-factor)
+                                                    (min 1) ; Can't match more than 100% of max-jobs-for-autoscaling.
+                                                    (* max-jobs-for-autoscaling)
+                                                    int
+                                                    (max number-unmatched-jobs)) ; Autoscale at least the pods that failed to match.
                 ;; We need to filter pending jobs based on quota so that we don't
                 ;; trigger autoscaling beyond what users have quota to actually run
                 autoscalable-jobs (->> pool-name
                                        (get @pool-name->pending-jobs-atom)
                                        (tools/filter-pending-jobs-for-quota pool-name (atom {}) (atom {})
-                                         user->quota user->usage (tools/global-pool-quota (config/pool-quotas) pool-name)))]
-            ;; This call needs to happen *after* launch-matched-tasks!
-            ;; in order to avoid autoscaling tasks taking up available
-            ;; capacity that was already matched for real Cook tasks.
-            (trigger-autoscaling! autoscalable-jobs pool-name compute-clusters job->acceptable-compute-clusters-fn)
+                                                                            user->quota user->usage (tools/global-pool-quota (config/pool-quotas) pool-name))
+                                       (take max-jobs-for-autoscaling-scaled))
+                filtered-autoscalable-jobs (remove #(.getIfPresent caches/recent-synthetic-pod-job-uuids (:job/uuid %)) autoscalable-jobs)]
+            ; When we have at least a minimum jobs being looked at, metric which fraction have matched. This lets us measure how well we're matching on existing resources.
+            ; We only measure when there's a minimum jobs being considered so that our measurements are less noisy.
```

Comment:
```suggestion
            ; When we have at least a minimum number of jobs being looked at, metric which fraction have matched. 
            ; This lets us measure how well we're matching on existing resources.
            ; We only measure when there's a minimum number of jobs being considered so that our measurements are less noisy.
```

Reply:
Done.